File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> st = new Stack<>() ; 

        for ( char ch : s.toCharArray()){
            if(ch == '(' || ch == '{' || ch == '['){
                st.push(ch) ; 
            }
            else {
                if ( st.isEmpty()) return false ; 
                char c = st.peek() ; 
                if( c == '(' && ch != ')') return false ; 
                else if ( c == '[' && ch != ']') return false ; 
                else if( c == '{' && ch != '}') return false ; 
                else {
                    st.pop() ; 
                }
            }
        }

        return st.isEmpty() ; 
    }
}

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        stack = [] 

        dict1 = { '(' : ')' , 
            '{' : '}' ,
            '[' : ']'}


        s = list(s) 

        for i in range(len(s)) :
            if s[i] in dict1.keys() :
                stack.append(s[i]) 
            else :
                if not stack :
                    return False 
                ch = stack[-1]
                if s[i] != dict1[ch] :
                    return False 
                else :
                    stack.pop()

        if len(stack) >= 1 :
            return False 

        return True 